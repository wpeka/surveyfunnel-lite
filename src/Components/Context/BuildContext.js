import React from 'react';
import update from 'immutability-helper';
import { ItemTypes } from '../../Data';
import fetchData from '../../HelperComponents/fetchData';
const { applyFilters } = wp.hooks;
export class BuildContextProvider extends React.Component {
	// buildContext state which contains List of questions, type, title, proActive or not.
    state = {
        List: {
			[ItemTypes.START_ELEMENTS]: [],
			[ItemTypes.CONTENT_ELEMENTS]: [],
            [ItemTypes.RESULT_ELEMENTS]: [],
		},
        title: '',
		type: '',
		proActive: false,
        outComeData:[],
    };

	// after component mounted get build data.
    componentDidMount() {
        const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_get_build_data',
            post_id
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData( ajaxURL, data )
        .then(data => {
            if ( data.data.build !== '' ) {
                let build = JSON.parse(data.data.build);
                let title = data.data.post_title;
				let type  = data.data.survey_type;
				let proActive = data.data.proActive;
                this.setState( {
                    ...build,
                   title,
				   type,
					proActive: proActive === '1' ? true : false
                } );
            }
            else {
                let title = data.data.post_title;
				let type  = data.data.survey_type;
                this.setState( {
                   title,
				   type
                } );
            }
        })
        this.getDataOutcome();
    }

	// add to list when question is newly created.
    addToList = (data) => {
		let newState = JSON.parse(JSON.stringify({...this.state}));

		newState.List[data.type].push(data);
        this.setState(newState);
    }

	// when question is being edited.
    editList = ( data ) => {
        let newState = JSON.parse(JSON.stringify({...this.state}));

        let newList = newState.List[data.type].slice();
        for(let i = 0; i<newList.length; i++) {
            if(data.id === newList[i].id) {
                newList[i] = data;
                break;
            }
        }
        newState.List[data.type] = newList;
        this.setState(newState);
    }

	// Card component uses this function in order to rearrage the cards.
    moveCard = (dragIndex, hoverIndex, data) => {
        const dragCard = this.state.List[data.type][dragIndex];
        const newList = this.state.List[data.type].slice();

        let newCardList = update(newList, {
            $splice: [
                [dragIndex, 1],
                [hoverIndex, 0, dragCard],
            ],
        });

        this.setState({
            List: {
                ...this.state.List,
                [data.type]: newCardList
            },
        });
    };

	// to delete item in the list.
    deleteItemInList = ( data ) => {
        let newState = JSON.parse(JSON.stringify({...this.state}));
        let newList = newState.List[data.type].slice();
        for(let i = 0; i<newList.length; i++) {
            if(data.id === newList[i].id) {
                newList.splice(i, 1);
                break;
            }
        }
        newList = applyFilters('deleteItemInList', newList, data);
        newState.List[data.type] = newList;
        this.setState(newState);
    }

	// generate unique id for each question.
    generateId = () => {   
        return Math.random().toString(36).substring(2) + (new Date()).getTime().toString(36);        
    }

	// saving data in the backend.
    saveData = (e) => {
        e.target.classList.add('surveyfunnel-lite-button-loading');
        const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
		const stateData = JSON.stringify( { ...this.state } );
		const hasProQuestions = stateData.match( /proVersionQuestionType/g ) ? true : false;
        const data = {
            state: JSON.stringify( { ...this.state } ),
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_save_build_data',
            post_id,
            post_title: this.state.title,
			hasProQuestions
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData(ajaxURL, data)
        .then(data => {
            e.target.classList.remove('surveyfunnel-lite-button-loading');
        });
    }
    //savinf outcome box data
    saveDataOutcome = (list) => {
        const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_save_build_data_outcome',
            post_id,
            list:JSON.stringify(list),
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData(ajaxURL, data)
        .then(data => {
            // console.log(x);
        });
        // this.setState({
        //     outComeData:list,
        // })
    }
    getDataOutcome = () => {
        console.log("i am result");
        const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_get_build_data_outcome',
            post_id,
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData(ajaxURL, data)
        .then(data => {
            console.log(data.data);
            this.setState({
                outComeData:data.data,
            })
        });
    }

	// to handle title state change.
    handleChangeTitle = ( title ) => {
        this.setState({
            title
        });
    }

    render() {
        return (
            <BuildContext.Provider
                value={{ ...this.state, addToList: this.addToList, getCount: this.getCount, editList: this.editList, deleteItemInList: this.deleteItemInList, moveCard: this.moveCard, generateId: this.generateId, saveData: this.saveData, handleChangeTitle: this.handleChangeTitle ,saveDataOutcome:this.saveDataOutcome}}
            >
                {this.props.children}
                
            </BuildContext.Provider>
        );
    }
}
  
export const BuildContext = React.createContext();