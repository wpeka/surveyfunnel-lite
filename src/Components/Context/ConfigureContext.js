import { createContext, Component, useState, useContext, useEffect } from "react";
const { applyFilters, doAction } = wp.hooks;
import fetchData from "../../HelperComponents/fetchData";

export function ConfigureContextProvider( props ) {

	// metaInfo state.
	const [metaInfo, setMetaInfo] = useState({
		title: '',
		description: ''
	});

	// companybranding state.
	const [companyBranding, setCompanyBranding] = useState(true);
	const [proActive, setProActive] = useState(true);
	const [proSettings, setProSettings] = useState({
		...applyFilters( 'configureProState', {} )
	});

	// get posts and pages state.
	const [options, setOptions] = useState([]);

	// useEffect hook to get configuration data when component is mounted.
	useEffect(() => {
		const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_get_configuration_data',
            post_id,
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData( ajaxURL, data )
        .then(data => {
            if ( data.data.configure === '' ) {
				return;
			}
            setProActive(data.data.proActive);
			let configure = JSON.parse(data.data.configure);
			setMetaInfo(configure.metaInfo);
			setCompanyBranding(configure.companyBranding);
			doAction( 'configureMount', configure, setProSettings );
        })

		const getOptions = () => {
			const ajaxSecurity = document.getElementById('ajaxSecurity').value;
			const post_id = new URLSearchParams(window.location.search).get('post_id');
			const data = {
				security: ajaxSecurity,
				action: 'surveyfunnel_lite_get_posts_pages',
				post_id,
				links: true,
			};
			const ajaxURL = document.getElementById('ajaxURL').value;
			fetchData( ajaxURL, data )
			.then(data => {
				setOptions(data.data);
			})
		}
	
		getOptions();
	}, []);

	// handle change in meta.
	const handleMetaChange = (e) => {
		setMetaInfo({
			...metaInfo,
			[e.target.name]: e.target.value,
		});
	}

	// save configuration data.
	const saveConfiguration = (e) => {
		e.target.classList.add('surveyfunnel-lite-button-loading');
		const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'surveyfunnel_lite_save_configuration_data',
            post_id,
			configuration: JSON.stringify({metaInfo, companyBranding, proSettings})
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData( ajaxURL, data )
        .then(data => {
			e.target.classList.remove('surveyfunnel-lite-button-loading');
        })
	}

	const value = {
		metaInfo, setMetaInfo, companyBranding, setCompanyBranding, options, setOptions, handleMetaChange, saveConfiguration, proActive, proSettings, setProSettings
	}
	return(
		<ConfigureContext.Provider
			value={value}
		>
			{props.children}
		</ConfigureContext.Provider>
	);
}

export const ConfigureContext = createContext();