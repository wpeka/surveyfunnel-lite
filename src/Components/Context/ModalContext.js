import { createContext, Component } from "react";
import { useState } from "react";
import React from 'react';

export function ModalContextProvider( props ) {

	// state to show or hide modal.
	const [ showModal, setShowModal ] = useState(false);	
	// state to set current element
	const [ currentElement, setCurrentElement ] = useState( {} );
	const getShowModal = () => {
		return showModal;
	}
	const value = {
		showModal, setShowModal, currentElement, setCurrentElement, getShowModal
	}
	return(
		<ModalContext.Provider
			value={value}
		>
			{props.children}
		</ModalContext.Provider>
	);
}

export const ModalContext = createContext();