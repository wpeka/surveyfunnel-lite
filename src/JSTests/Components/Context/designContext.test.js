import React from 'react';
import { mount, shallow } from 'enzyme';
import {DesignContextProvider} from '../../../Components/Context/DesignContext';
import renderer from 'react-test-renderer';
// import DesignSettings from "../../../Components/Design/DesignSettings";

test("DesignContextProvider is rendering",()=>{
    const designWrapper=renderer.create(<DesignContextProvider />)
    const tree=designWrapper.toJSON();
    expect(tree).toMatchSnapshot();

})