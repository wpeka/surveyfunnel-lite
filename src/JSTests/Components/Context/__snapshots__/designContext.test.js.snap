// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesignContextProvider is rendering 1`] = `null`;
